artifactory/commands/replication/template.go

```diff
@@ -0,0 +1,207 @@
+		MapKey:       "",
+		Callback:     nil,
+	},
+	JobType: {
+		Options: []prompt.Suggest{
+			{Text: Pull, Description: "Pull replication"},
+			{Text: Push, Description: "Push replication"},
+		},
+		Msg:          "Select job type",
+		PromptPrefix: ">",
+		AllowVars:    true,
+		Writer:       nil,
+		MapKey:       "",
+		Callback:     jobTypeCallback,
+	},
+	RepoKey: {
+		Msg:          "Enter repo key",
+		PromptPrefix: ">",
+		AllowVars:    true,
+		Writer:       utils.WriteStringAnswer,
+		MapKey:       RepoKey,
+		Callback:     nil,
+	},
+	ServerId: {
+		Msg:          "Enter server id",
+		PromptPrefix: ">",
+		AllowVars:    true,
+		Writer:       utils.WriteStringAnswer,
+		MapKey:       ServerId,
+		Callback:     nil,
+	},
+	CronExp: {
+		Msg:          "Enter cron expression",
+		PromptPrefix: ">",
+		AllowVars:    true,
+		Writer:       utils.WriteStringAnswer,
+		MapKey:       CronExp,
+		Callback:     nil,
+	},
+	EnableEventReplication: utils.BoolQuestionInfo,
+	Enabled:                utils.BoolQuestionInfo,
+	SyncDeletes:            utils.BoolQuestionInfo,
+	SyncProperties:         utils.BoolQuestionInfo,
+	SyncStatistics:         utils.BoolQuestionInfo,
+	PathPrefix: {
+		Msg:          "Enter path prefix",
+		PromptPrefix: ">",
+		AllowVars:    false,
+		Writer:       utils.WriteStringAnswer,
+		MapKey:       PathPrefix,
+		Callback:     nil,
+	},
+	SocketTimeoutMillis: {
+		Msg:          "Enter socket timeout millis",
+		PromptPrefix: ">",
+		AllowVars:    false,
+		Writer:       utils.WriteStringAnswer,
+		MapKey:       SocketTimeoutMillis,
+		Callback:     nil,
+	},
+}
+
+func jobTypeCallback(iq *utils.InteractiveQuestionnaire, jobType string) (string, error) {
+	iq.MandatoryQuestionsKeys = append(iq.MandatoryQuestionsKeys, CronExp)
+	if jobType == Pull {
+		iq.OptionalKeysSuggests = getAllPossibleOptionalRepoConfKeys()
+	} else {
+		iq.MandatoryQuestionsKeys = append(iq.MandatoryQuestionsKeys, ServerId)
+		iq.OptionalKeysSuggests = getAllPossibleOptionalRepoConfKeys()
+	}
+	return "", nil
+}
+
+func getAllPossibleOptionalRepoConfKeys(values ...string) []prompt.Suggest {
+	optionalKeys := []string{utils.SaveAndExit, Enabled, SyncDeletes, SyncProperties, SyncStatistics, PathPrefix, EnableEventReplication, SocketTimeoutMillis}
+	if len(values) > 0 {
+		optionalKeys = append(optionalKeys, values...)
+	}
+	return utils.GetSuggestsFromKeys(optionalKeys, suggestionMap)
+}
+
+func optionalKeyCallback(iq *utils.InteractiveQuestionnaire, key string) (value string, err error) {
+	if key != utils.SaveAndExit {
+		valueQuestion := iq.QuestionsMap[key]
+		// Since we are using default question in most of the cases we set the map key here
+		valueQuestion.MapKey = key
+		valueQuestion.PromptPrefix = InsertValuePromptMsg + key
+		if valueQuestion.Options != nil {
+			valueQuestion.PromptPrefix += utils.PressTabMsg
+		}
+		valueQuestion.PromptPrefix += " >"
+		value, err = iq.AskQuestion(valueQuestion)
+	}
+	return value, err
+}
```

Comment:
This code repeated in repository/template.go
I suggest moving the function to questionnaire.go